src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.go

```diff
@@ -171,13 +172,20 @@ func (st *StoragePowerActorState_I) _sampleMinersToSurprise(rt Runtime, challeng
 		index++
 	}
 
-	sampledMiners := make([]addr.Address, challengeCount)
+	sampledMiners := make([]addr.Address, 0)
+
 	for chall := 0; chall < challengeCount; chall++ {
 		minerIndex := filproofs.TicketToRandomInt(ticket.Output(), chall, ptSize)
 		panic(minerIndex)
-		// hack
+		// hack to turn bigint into int
 		minerIndexInt := 0
-		sampledMiners = append(sampledMiners, allMiners[minerIndexInt])
+		potentialChallengee := allMiners[minerIndexInt]
```

Comment:
Let's send all the challenges and handle this at StorageMinerActor to avoid the round trip. Will make the change.